energy-integration/common-modules/weekly-rewards-splitting/src/global_info.rs

```diff
@@ -40,27 +43,55 @@ pub trait WeeklyRewardsGlobalInfo: crate::events::WeeklyRewardsSplittingEventsMo
             last_global_update_mapper.set(current_week);
         }
 
-        let total_locked_tokens_mapper = self.total_locked_tokens_for_week(current_week);
-        total_locked_tokens_mapper.update(|total_locked| {
-            *total_locked -= prev_user_energy.get_total_locked_tokens();
-            *total_locked += current_user_energy.get_total_locked_tokens();
-        });
-
-        let total_energy_mapper = self.total_energy_for_week(current_week);
-        total_energy_mapper.update(|total_energy| {
-            // revert the 7 * tokens removed in global decrease step
-            if user_last_active_week != current_week {
-                *total_energy += prev_user_energy.get_total_locked_tokens() * EPOCHS_IN_WEEK;
-            }
+        let total_removed_from_buckets = self.shift_buckets_and_get_removed_token_amount();
```

Comment:
function is a little bit big. 94 lines - if you could make a sub function for lines 46-75 it would be awesome :).

Reply:
Done, refactored into 3 separate functions.